lib/matplotlib/axes/_base.py

```diff
@@ -4108,19 +4108,45 @@ def pick(self, *args):
         martist.Artist.pick(self, args[0])
 
     def get_default_bbox_extra_artists(self):
+        """
+        Return a default list of artists that are used for the bounding box
+        calculation.  Artists are excluded either by not being visible
+        or ``artist.set_in_layout(False)``.
+        """
         return [artist for artist in self.get_children()
-                if artist.get_visible()]
+                if (artist.get_visible() and artist.get_in_layout())]
 
-    def get_tightbbox(self, renderer, call_axes_locator=True):
+    def get_tightbbox(self, renderer, call_axes_locator=True,
+            bbox_extra_artists=None):
         """
-        Return the tight bounding box of the axes.
-        The dimension of the Bbox in canvas coordinate.
+        Return the tight bounding box of the axes, including axis and their
+        decorators (xlabel, title, etc).
+
+        Artists that have ``artist.set_in_layout(False)`` are not included
+        in the bbox.
 
-        If *call_axes_locator* is *False*, it does not call the
-        _axes_locator attribute, which is necessary to get the correct
-        bounding box. ``call_axes_locator==False`` can be used if the
-        caller is only intereted in the relative size of the tightbbox
-        compared to the axes bbox.
+        Parameters
+        ----------
+        renderer : `.RendererBase` instance
+            renderer that will be used to draw the figures (i.e.
+            ``fig.canvas.get_renderer()``)
+
+        bbox_extra_artists : list of `.Artist` or ``None``
+            List of artists to include in the tight bounding box.  If
+            ``None`` (default), then all artist children of the axes are
+            included in the tight bounding box.
+
+        call_axes_locator : boolean (default ``True``)
+            If *call_axes_locator* is ``False``, it does not call the
+            ``_axes_locator`` attribute, which is necessary to get the correct
+            bounding box. ``call_axes_locator=False`` can be used if the
+            caller is only interested in the relative size of the tightbbox
+            compared to the axes bbox.
+
+        Returns
+        -------
+        bbox : `.BboxBase`
+            containing the bounding box (in figure pixel co-ordinates).
```

Comment:
I think this is OK: From the Transforms Tutorial: "pixel coordinate system of the display", but thats a bit wrong because we never deal w/ the pixels co-ordinate system of the display, but rather the pixels of the figure window (viewport?)